src/Operations.jl

```diff
@@ -46,8 +46,21 @@ end
 is_dep(ctx::Context, pkg::PackageSpec) =
     any(uuid -> uuid == pkg.uuid, [uuid for (name, uuid) in ctx.env.project.deps])
 
-function load_direct_deps!(ctx::Context, pkgs::Vector{PackageSpec}; version::Bool=true)
-    # load rest of deps normally
+#TODO rename
+function load_version(version, fixed, preserve::PreserveLevel)
+    version === nothing && return VersionSpec() # for stdlibs
+    fixed && return version
+    preserve == PRESERVE_ALL && return something(version, VersionSpec())
+    preserve == PRESERVE_DIRECT && return something(version, VersionSpec())
+    if preserve == PRESERVE_SEMVER && version != VersionSpec()
+        return Types.semver_spec("$(version.major)")
```

Comment:
This doesn't seem completely correct? E.g. if the version is `0.4.2`, calling  `semver_spec("0")` wouldn't be right?

Reply:
I actually wasn't sure about this. Looking at https://semver.org/, the convention seems to be that minor version changes are considered breaking changes? I'll fix it

Reply:
Isn't it enough to call `Types.semver_spec("$(version.major).($version.minor).($version.patch)")`?